src/sqlfluff/rules/L003.py

```diff
@@ -776,136 +660,358 @@ def _process_current_line(
-                        fixes=fixes,
-                    )
+                fixes += self._coerce_indent_to(
+                    desired_indent="".join(
+                        elem.raw for elem in this_line.indent_buffer
+                    ),
+                    current_indent_buffer=prev_line.indent_buffer,
+                    current_anchor=anchor,
+                )
 
-            # This was a valid comparison, so if it doesn't flag then
-            # we can assume that we're ok.
-            self.logger.debug("    Indent deemed ok comparing to #%s", k)
-
-            # Given that this line is ok, consider if the preceding lines are
-            # comments. If they are, lint the indentation of the comment(s).
-            fixes = []
-            for n in range(this_line_no - 1, -1, -1):
-                if n in memory["comment_lines"]:
-                    # The previous line WAS a comment.
-                    prev_line = res[n]
-                    if this_line["indent_size"] != prev_line["indent_size"]:
-                        # It's not aligned.
-                        # Find the anchor first.
-                        anchor: BaseSegment = None  # type: ignore
-                        for seg in prev_line["line_buffer"]:
-                            if seg.is_type("comment"):
-                                anchor = seg
-                                break
-                        # Make fixes.
-                        fixes += self._coerce_indent_to(
-                            desired_indent="".join(
-                                elem.raw for elem in this_line["indent_buffer"]
-                            ),
-                            current_indent_buffer=prev_line["indent_buffer"],
-                            current_anchor=anchor,
-                        )
+                memory.problem_lines.add(n)
+
+        if not fixes:
+            return None
+
+        return LintResult(
+            anchor=anchor,
+            memory=memory,
+            description="Comment not aligned with following line.",
+            fixes=fixes,
+        )
+
+    def _handle_hanging_indents(
+        self,
+        this_line: _LineSummary,
+        previous_lines: List[_LineSummary],
+        memory: _Memory,
+    ) -> Optional[LintResult]:
+        if len(previous_lines) == 0:
+            return None
+
+        last_line = _find_last_meaningful_line(previous_lines)
+        if not last_line:
+            return None
+        # Handle Hanging Indents
+        is_anchor_indent_match = (
+            this_line.anchor_indent_balance == last_line.anchor_indent_balance
+        )
+        is_end_indent_match = this_line.indent_size == last_line.indent_size
+        is_known_hanging_line = last_line.line_no in memory.hanging_lines
+        # There MUST also be a non-zero indent. Otherwise we're just on the
+        # baseline.
+        if not this_line.indent_size > 0:
```

Comment:
This comparison looks strange. Can we clean this up to avoid having `not` and `>` in the same expression?

Reply:
Funny black doesn't complain I was just quickly trying to invert things into guard statements